File: apps/workspace-engine/test/integration/creators/user_approval_record.go
package creators

import (
	"time"
	"workspace-engine/pkg/oapi"

	"github.com/google/uuid"
)

// NewUserApprovalRecord creates a test UserApprovalRecord with sensible defaults
// All fields can be overridden directly after creation
func NewUserApprovalRecord(versionId, environmentId, userId string) *oapi.UserApprovalRecord {
	return &oapi.UserApprovalRecord{
		VersionId:     versionId,
		EnvironmentId: environmentId,
		UserId:        userId,
		Status:        oapi.ApprovalStatusApproved,
		Reason:        nil,
		CreatedAt:     time.Now().Format(time.RFC3339),
	}
}

// NewUserID generates a new UUID for test users
func NewUserID() string {
	return uuid.New().String()
}

File: packages/db/src/schema/user-approval-record.ts
import {
  pgTable,
  primaryKey,
  text,
  timestamp,
  uuid,
} from "drizzle-orm/pg-core";

export const userApprovalRecord = pgTable(
  "user_approval_record",
  {
    versionId: uuid("version_id").notNull(),
    userId: uuid("user_id").notNull(),
    environmentId: uuid("environment_id").notNull(),
    status: text("status").notNull(),
    reason: text("reason"),
    createdAt: timestamp("created_at", { withTimezone: true })
      .notNull()
      .defaultNow(),
  },
  (t) => [primaryKey({ columns: [t.versionId, t.userId, t.environmentId] })],
);
